master/internal/rm/agentrm/agent_resource_manager.go

```diff
@@ -325,7 +325,7 @@ func (a *agentResourceManager) Receive(ctx *actor.Context) error {
 		ctx.Respond(response.Get())
 
 	case sproto.GetJobQ:
-		if msg.ResourcePool == "" {
+		if msg.ResourcePool == "" && !a.config.NoDefaultResourcePools {
 			msg.ResourcePool = a.config.DefaultComputeResourcePool
```

Comment:
this doesn't seem quite right. this was intended to default the resource pool if none was passed, but if there are no defaults asking `ctx.Child("")` doesn't make much sense. so maybe we need to just return an error here, but I would check what API the callers expect, too.

Reply:
I thought about this vs just letting the actor error - the returned error is that the resource pool "" is not found, which I thought made sense, but I'll make it clearer

Reply:
I would just make that error case clearer. I didn't notice that it fell through and got a not found anyway. Failing the actor doesn't make sense to me.

Reply:
TBH, with the rest of the actor system, i wouldn't have been surprised if ctx.Child("") panicked before i read the code lol